### ISSUE TITLE
JDocument does not support Open Graph metadata

### ISSUE THREAD

**Issue by mbabker** [joomla/joomla-cms#10671]
#### Steps to reproduce the issue

Add open graph metadata using `JDocument::setMetaData()`

```php
$document->setMetaData('og:image', 'https://cdn.joomla.org/images/Joomla_logo.png');
```

#### Expected result

When rendering, the resulting meta tag per http://ogp.me/ should be:

```html
<meta property="og:image" content="https://cdn.joomla.org/images/Joomla_logo.png" />
```

#### Actual result

When rendering, the resulting meta tag is:

```html
<meta name="og:image" content="https://cdn.joomla.org/images/Joomla_logo.png" />
```

#### Additional comments

The use of `JDocument::addCustomTag()` is required to mitigate this due to the lack of native support.

**Comment by n9iels** [comment]
We could add a new parameter `JDocument::setMetaData()` that can be set to `true` for open graph. (See this file and line https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/document/document.php#l424).

And for the head renderer a check to generate the correct html (see this file and line https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/document/renderer/html/head.php#L92)

**Comment by andrepereiradasilva** [comment]
you also need to add the open graph namespace in the hmtl tag right?

**Comment by mbabker** [comment]
I honestly don't know but if so that presents another issue in that there
is absolutely no way without an onAfterRender plugin to set stuff on the
base HTML tag.

**Comment by andrepereiradasilva** [comment]
we would need some sort of an jdoc include in the template for that
Something like:
```html
<html lang="<?php echo $this->language; ?>" dir="<?php echo $this->direction; ?>"<jdoc:include type="namespaces" />>
```
And them make JDocument render those with something like
```php
$doc->addHtmlNameSpace($attribute, $value);

$doc->addHtmlNameSpace("prefix", "og: http://ogp.me/ns#");
$doc->addHtmlNameSpace("prefix", "fb: http://ogp.me/ns/fb#");
```


So it can render
```html
<!DOCTYPE html>
<html lang="en-gb" dir="ltr" prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#">
```
Since html is extensible trough prefixes/namespaces it would be godd if joomla could deal with this.

**Comment by Bakual** [comment]
In my extension it seems to work without the namespace and the Facebook debug tool also doesn't complain about the missing namespace.

**Comment by andrepereiradasilva** [comment]
yes it works, but it's not valid HTML code if you don't add the namespace right?
See http://ogp.me

**Comment by Bakual** [comment]
The namespace isn't needed for it to be valid HTML. OpenGraph uses regular HTML elements (namely the `<meta>` tag). It's only the values which define the OpenGraph functionality.

**Comment by andrepereiradasilva** [comment]
you're right, just test it in the validator (https://validator.w3.org/#validate_by_input) and says it's valid HTML witout the prefix.
```
<!DOCTYPE html>
<html>
<head>
<title>Test</title>
<meta property="og:title" content="The Rock" />
<meta property="og:type" content="video.movie" />
<meta property="og:url" content="http://www.imdb.com/title/tt0117500/" />
<meta property="og:image" content="http://ia.media-imdb.com/images/rock.jpg" />
</head>
<body>
<p>test</p>
</body>
</html>
```

IIRC sometime ago the validator add issues with that.
But since it nows says it's valid i think you're right.

So it seems for this case you don't need the namespace.

**Comment by andrepereiradasilva** [comment]
@mbabker please test https://github.com/joomla/joomla-cms/pull/10682

**Comment by brianteeman** [comment]
Closed as we have a a PR #10682 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10671">issues.joomla.org/joomla-cms/10671</a>.</sub>
